cmd/ci-operator-configresolver/main.go

```diff
@@ -39,46 +39,25 @@ type options struct {
 	flatRegistry bool
 }
 
-type traceResponseWriter struct {
-	http.ResponseWriter
-	statusCode int
-	size       int
-}
-
-func (w *traceResponseWriter) WriteHeader(code int) {
-	w.statusCode = code
-	w.ResponseWriter.WriteHeader(code)
-}
-
-func (w *traceResponseWriter) Write(data []byte) (int, error) {
-	size, err := w.ResponseWriter.Write(data)
-	w.size += size
-	return size, err
-}
-
 var (
-	configresolverMetrics = struct {
-		httpRequestDuration *prometheus.HistogramVec
-		httpResponseSize    *prometheus.HistogramVec
-		errorRate           *prometheus.CounterVec
-	}{
-		httpRequestDuration: prometheus.NewHistogramVec(
+	configresolverMetrics = httphelper.Metrics{
```

Comment:
Can we add a constructor for this that just takes the metrics `namespace` as argument? Something like this
```
func NewMetrics(namespace string) Metrics {
 return Metrics{
  HttpRequestDuration: prometheus.NewHistogramVec(
prometheus.HistogramOpts{
 Namespace: namespace,
 Name: "http_request_duration_seconds",
  Help:    "http request duration in seconds",
Buckets: []float64{0.0005, 0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2}

}
  )
}
```

(weirdly formatted, sorry).

The client lib prefixes the `Name` with `Namespace + "_"` if the latter is set, hence this should not result in a namechange of the metric and we avoid duplicating this code. This is espcially important since the client panics if you use labels that were not defined on the metrics

Reply:
And are we ok with stuff like duration buckets being common for different applications?

Reply:
Yes, there is no reason at all for them to be different.